### ISSUE TITLE
Battebit Remastered Support

### ISSUE THREAD

**Issue by JesseUCAVed** [gamedig/node-gamedig#530]
It would be great if the game Battlebit Remastered was added to the supported games list.
I currently have to use the BattleMetrics API for pulling info, which isn't ideal when this fantastic library/plugin exists. :)

**Comment by podrivo** [comment]
Hey @JesseUCAVed!

Do you have an IP:PORT so we can test?
I wasn't able to find server IPs on [BattleMetrics](https://www.battlemetrics.com/servers/battlebit).

**Comment by JesseUCAVed** [comment]
Hey,

First time with this Github thing so not sure if replying to this e-mail
works.

IP for my server is 104.128.50.35 and I know one of the ports(primary) is
29993 and I think the other is 29994 or 29992. I can verify the last two
later for sure.

Thanks

**Comment by podrivo** [comment]
Thank you!

I've tried running the attempt protocol tool in these IPs and PORTs, but it failed on all protocols.
We might need more info about how to query BatteBit Remastered servers.

**Comment by JesseUCAVed** [comment]
There is the API endpoint port of 29295, but that's more for management I
believe.
Game port is definitely 29993. I'm just trying to remember if it had a
seperate query port.
I know Battlemetrics somehow pulls all the data, but they always have some
magical way of doing that and pulling more than Steam generally allows.

**Comment by podrivo** [comment]
Tested with the IP `104.128.50.35` on ports `29295` and `29993`, but no luck... :(

Found this other server list, but no IPs again. https://bbspy.net/
Server info might be via RCON.

**Comment by podrivo** [comment]
Found a public API that queries all servers: https://publicapi.battlebit.cloud/Servers/GetServerList
Again, no IPs on this list.

@JesseUCAVed Can you find your server in the link above?

**Comment by JesseUCAVed** [comment]
My Battlebit server was listed before I added RCON. Battlebit doesn't
natively have RCON so for battlemetrics it's "hacked" in so for that other
site to be retrieving it, it has to be public somewhere. 🤔

**Comment by JesseUCAVed** [comment]
I will look shortly after dinner.

**Comment by podrivo** [comment]
Yeah, server info might be coming from the public API.
I haven't checked everything, but I bet it's the same info in [BattleMetrics](https://www.battlemetrics.com/servers/battlebit), this [server list](https://bbspy.net/) and the [public API](https://publicapi.battlebit.cloud/Servers/GetServerList).

**Comment by JesseUCAVed** [comment]
Yeah my server is in that list automatically and pulling the same data that
Battlemetrics does.

**Comment by JesseUCAVed** [comment]
If you and anyone else involved in the development/addition of Battlebit
wants you can join our Discord to communicate with myself/my clan dev
easier.
https://discord.gg/rsclan

Just le me know if you do so I can grant you access to a channel for this
discussion.

**Comment by podrivo** [comment]
Nice! So we have a public API that queries all servers: https://publicapi.battlebit.cloud/Servers/GetServerList
But no IP or PORT to have a direct connection to a specific server.

I'm not sure how to implement this, but maybe instead of an IP:PORT we could pass a NAME that will filter the object got from the public API?

Something like:

```
const url = 'https://publicapi.battlebit.cloud/Servers/GetServerList'
const searchName = 'Frontline'

fetch(url)
  .then(response => response.json())
  .then(data => {
    const filteredResults = data.filter(item => item.Name.includes(searchName))
    console.log(filteredResults)
  })
  .catch(error => console.error('Error fetching data:', error))
```

```
[
    {
        "Name": "#10 [E4GL] | Frontline | discord.gg/E4GL",
        "Map": "Wakistan",
        "MapSize": "Ultra",
        "Gamemode": "FRONTLINE",
        "Region": "Europe_Central",
        "Players": 0,
        "QueuePlayers": 0,
        "MaxPlayers": 254,
        "Hz": 120,
        "DayNight": "Day",
        "IsOfficial": false,
        "HasPassword": false,
        "AntiCheat": "EAC",
        "Build": "Production 2.2.5 Hotfix"
    },
    {
        "Name": " [01][Clã BR] SemChorar.com [Frontline 24/7]",
        "Map": "SandySunset",
        "MapSize": "Ultra",
        "Gamemode": "FRONTLINE",
        "Region": "Brazil_Central",
        "Players": 120,
        "QueuePlayers": 2,
        "MaxPlayers": 254,
        "Hz": 144,
        "DayNight": "Day",
        "IsOfficial": false,
        "HasPassword": false,
        "AntiCheat": "EAC",
        "Build": "Production 2.2.5 Hotfix"
    }
]
```

**Comment by JesseUCAVed** [comment]
That's a crazy bit of info to have without an ip to narrow it lol

**Comment by livingflore** [comment]
> Nice! So we have a public API that queries all servers: https://publicapi.battlebit.cloud/Servers/GetServerList But no IP or PORT to have a direct connection to a specific server.
> 
> I'm not sure how to implement this, but maybe instead of an IP:PORT we could pass a NAME that will filter the object got from the public API?
> 
> Something like:
> 
> ```
> const url = 'https://publicapi.battlebit.cloud/Servers/GetServerList'
> const searchName = 'Frontline'
> 
> fetch(url)
>   .then(response => response.json())
>   .then(data => {
>     const filteredResults = data.filter(item => item.Name.includes(searchName))
>     console.log(filteredResults)
>   })
>   .catch(error => console.error('Error fetching data:', error))
> ```
> 
> ```
> [
>     {
>         "Name": "#10 [E4GL] | Frontline | discord.gg/E4GL",
>         "Map": "Wakistan",
>         "MapSize": "Ultra",
>         "Gamemode": "FRONTLINE",
>         "Region": "Europe_Central",
>         "Players": 0,
>         "QueuePlayers": 0,
>         "MaxPlayers": 254,
>         "Hz": 120,
>         "DayNight": "Day",
>         "IsOfficial": false,
>         "HasPassword": false,
>         "AntiCheat": "EAC",
>         "Build": "Production 2.2.5 Hotfix"
>     },
>     {
>         "Name": " [01][Clã BR] SemChorar.com [Frontline 24/7]",
>         "Map": "SandySunset",
>         "MapSize": "Ultra",
>         "Gamemode": "FRONTLINE",
>         "Region": "Brazil_Central",
>         "Players": 120,
>         "QueuePlayers": 2,
>         "MaxPlayers": 254,
>         "Hz": 144,
>         "DayNight": "Day",
>         "IsOfficial": false,
>         "HasPassword": false,
>         "AntiCheat": "EAC",
>         "Build": "Production 2.2.5 Hotfix"
>     }
> ]
> ```

hi, BBR staff here
we do not have any query for servers except this public method unfortunately
i will consult with Oki (lead dev) if we could add some kind of public server identifier to make it easier, but can't provide exact ETA of it (likely after upcoming update)

**Comment by JesseUCAVed** [comment]
Best bet is to just add an ip and port identifier to the public json. :)

**Comment by podrivo** [comment]
Hey @livingflore. Thanks for joining the conversation! (:

> Best bet is to just add an ip and port identifier to the public json. :)

Yes, this might do the trick! 👍

**Comment by xCausxn** [comment]
They could also look to add a one way hash of the ip:port, should keep it somewhat secure while also allowing easy querying.

**Comment by CosminPerRam** [comment]
Hey @livingflore, any updates?

**Comment by livingflore** [comment]
> Hey @livingflore, any updates?

unfortunately not, Oki is too busy with the update, but i'm def keeping this in my list

**Comment by JesseUCAVed** [comment]
> > Hey [@livingflore](https://github.com/livingflore), any updates?
> 
> unfortunately not, Oki is too busy with the update, but i'm def keeping this in my list

Any updates to this update?
